operator/cmd/mesh/install.go

```diff
@@ -162,6 +162,16 @@ func Install(rootArgs *RootArgs, iArgs *InstallArgs, logOpts *log.Options, stdOu
 	if err != nil {
 		return fmt.Errorf("fetch Istio version: %v", err)
 	}
+
+	// return warning if current date is near the EOL date
+	t := time.Now()
+	warnMarker := color.New(color.FgYellow).Add(color.Italic).Sprint("WARNING:")
+	if t.Year() > operatorVer.OperatorEOLYear {
+		fmt.Printf("%s You are installing a EOL version, see https://istio.io/latest/docs/releases/supported-releases/ for supported release\n", warnMarker)
+	} else if t.Year() == operatorVer.OperatorEOLYear && t.Month() >= operatorVer.OperatorEOLMonth {
+		fmt.Printf("%s may be installing an EOL version: see https://istio.io/latest/docs/releases/supported-releases/ for supported releases\n", warnMarker)
```

Comment:
I think we should just always say "MAY". The reason is we could retroactively extend support for a version.

Also the month vs year check is not really stable; we could EOL on Jan 2021 and we would warn "maybe" for 11 months, or on Nov 2021 and we would only warn for 1 month?

Reply:
can you elaborate more on

`Also the month vs year check is not really stable; we could EOL on Jan 2021 and we would warn "maybe" for 11 months, or on Nov 2021 and we would only warn for 1 month?`

I don't get the idea

Reply:
I assume the intent was that `t.Year() > operatorVer.OperatorEOLYear` check means "You are really far EOL" and the next check is "you maybe JUST became EOL".

if so, its better to check the exact delta (e.g 6 months after expected EOL) instead of just the year.

However, I think we should always say MAY anyways so we don't need the check at all.

Reply:
sure, always say `MAY` is preferred here due to changes of extended support.